baybe/targets/numerical.py

```
@@ -573,6 +590,41 @@ def clamp(
         max = max if max is not None else float("inf")
         return self._append_transformation(ClampingTransformation(min, max))
 
+    def _hold_output(
+        self, abscissa: float, direction: MatchMode | str, /
+    ) -> NumericalTarget:
+        """Hold the target value beyond a certain abscissa value."""
+        direction = MatchMode(direction)
+
+        if direction is MatchMode.EQ:
```

Why not `if ... elif ... else`? Wouldn't that avoid the `assert_never`?

---

There is a conceptual difference. `assert_never` is the type-safe alternative to your approach, it ensure that all options have been exhaustively evaluated. Assume that, for whatever reason, you add another member to `MatchMode`: your approach would, in the worst case, silently fail if you forget to adjust the code in all places where `MatchMode` is used, while the `assert_never` version will alarm you

---

closing becasue of the discussed advantages of the current code